dotnet/src/Planners/Planners.Core/Handlebars/Models/HandlebarsParameterTypeMetadata.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+#pragma warning disable IDE0130 // Namespace does not match folder structure
+namespace Microsoft.SemanticKernel.Planning.Handlebars.Models;
+#pragma warning restore IDE0130
+
+internal class HandlebarsParameterTypeMetadata
+{
+    [JsonPropertyName("name")]
+    public string Name { get; set; } = string.Empty;
+
+    [JsonPropertyName("isComplexType")]
+    public bool IsComplex { get; set; } = false;
+
+    /// <summary>
+    /// If this is a complex type, this will contain the properties of the complex type.
+    /// </summary>
+    [JsonPropertyName("properties")]
+    public List<SKParameterMetadata> Properties { get; set; } = new();
+
+    // Override the Equals method to compare the property values
+    public override bool Equals(object obj)
+    {
+        // Check if the obj is null or of a different type
+        if (obj == null || obj.GetType() != this.GetType())
+        {
+            return false;
+        }
+
+        // Cast the obj to HandlebarsParameterTypeMetadata
+        var other = (HandlebarsParameterTypeMetadata)obj;
+
+        // Compare the Name and IsComplex properties
+        if (this.Name != other.Name || this.IsComplex != other.IsComplex)
+        {
+            return false;
+        }
+
+        // Compare the Properties lists using a helper method
+        return ArePropertiesEqual(this.Properties, other.Properties);
+    }
+
+    // A helper method to compare two lists of SKParameterMetadata
+    private static bool ArePropertiesEqual(List<SKParameterMetadata> list1, List<SKParameterMetadata> list2)
+    {
+        // Check if the lists are null or have different lengths
+        if (list1 == null || list2 == null || list1.Count != list2.Count)
+        {
+            return false;
+        }
+
+        // Compare the elements of the lists by comparing the Name and ParameterType properties
+        for (int i = 0; i < list1.Count; i++)
+        {
+            if (!list1[i].Name.Equals(list2[i].Name, System.StringComparison.Ordinal) || !list1[i].ParameterType!.Equals(list2[i].ParameterType))
+            {
+                return false;
+            }
+        }
+
+        // If all elements are equal, return true
+        return true;
```

Comment:
The code can be replaced by `return this.Properties.SequenceEqual(other.Properties);` if SKParameterMetadata class implements Equals method and if sequential comparison is what we need here.

Reply:
We do need sequential property comparison but only need to check Name and PropertyType from the SKParameterMetadata object, since those are the only 2 properties we use in the HandlebarsParameterTypeMetadata object, 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/1ec6e66a-bdd7-4ae3-b876-1ca490848e35)

Also, SKParameterMetadata doesn't override the Equals method.

Reply:
> Also, SKParameterMetadata doesn't override the Equals method.

We could make it do so if desirable.

Reply:
This scenario is pretty specific to the Planners and how we render complex types for the function manual, but I think it could be beneficial to centralize this in the SKParameterMetadata object anyway (for future planners or data checks when sharing variables across multiple functions of the same plugin)